src/components/Account/AccountProfileContainer.tsx

```diff
@@ -0,0 +1,255 @@
+  showNotification,
+  updateProfile: profileActions.updateProfile
+};
+
+type IProps = IStateProps & IDispatchProps;
+
+interface IState {
+  genCount: number;
+  ethCount: number;
+}
+
+interface FormValues {
+  description: string;
+  githubURL: string;
+  name: string;
+}
+
+class AccountProfileContainer extends React.Component<IProps, IState> {
+
+  constructor(props: IProps) {
+    super(props);
+
+    this.state = {
+      ethCount: null,
+      genCount: null
+    };
+  }
+
+  public async componentWillMount() {
+    const { accountAddress, dao } = this.props;
+    const web3 = await Arc.Utils.getWeb3();
+    const getBalance = promisify(web3.eth.getBalance);
+    const ethBalance = await getBalance(accountAddress);
+
+    let votingMachineInstance: Arc.GenesisProtocolWrapper;
+    if (dao) {
+      const contributionRewardInstance = await Arc.ContributionRewardFactory.deployed();
+      const votingMachineAddress = (await contributionRewardInstance.getSchemeParameters(dao.avatarAddress)).votingMachineAddress;
+      votingMachineInstance = await Arc.GenesisProtocolFactory.at(votingMachineAddress);
+    } else {
+      votingMachineInstance = await Arc.GenesisProtocolFactory.deployed();
+    }
+    const stakingTokenAddress = await votingMachineInstance.contract.stakingToken();
+    const stakingToken = await (await Arc.Utils.requireContract("StandardToken")).at(stakingTokenAddress) as any;
+    const genBalance = await stakingToken.balanceOf(accountAddress);
+
+    this.setState({ ethCount: Util.fromWei(ethBalance), genCount: Util.fromWei(genBalance)});
+  }
+
+  public copyAddress = () => {
+    const { showNotification, accountAddress } = this.props;
+    Util.copyToClipboard(accountAddress);
+    showNotification(NotificationStatus.Success, `Copied to clipboard!`);
+  }
+
+  public async handleSubmit(values: FormValues, { props, setSubmitting, setErrors }: any ) {
+    const { accountAddress, updateProfile } = this.props;
+
+    const web3 = await Arc.Utils.getWeb3();
+    const text = "Please sign in to Alchemy";
+    const msg = ethUtil.bufferToHex(Buffer.from(text, 'utf8'));
+    const fromAddress = this.props.accountAddress;
+
+    let signature = localStorage.getItem("signature-" + fromAddress);
+    if (!signature) {
+      const method = 'personal_sign';
+      const sendAsync = promisify(web3.currentProvider.sendAsync);
+      const params = [msg, fromAddress];
+      const result = await sendAsync({ method, params, fromAddress });
+      signature = result.result;
+      localStorage.setItem("signature-" + fromAddress, signature);
+    }
+
+    const recoveredAddress = sigUtil.recoverPersonalSignature({ data: msg, sig: signature });
+
+    if (recoveredAddress == this.props.accountAddress) {
+      await updateProfile(accountAddress, values.name, values.description, signature);
+    } else {
+      console.error("Signing in failed, please ensure you are logged in to MetaMask with an address that you own");
```

Comment:
What are you "signing in" for here? What does "logged in to metamask with an address that you own" mean?

Reply:
its just using language people are more familiar with, but I agree it doesn't really map the flow we are presenting. I will change it.